### PR TITLE
feature(tool): Implementation of the Appsettings files linter

### PR DESCRIPTION
**Pull request by rgcouto**
### Appsetings Linter implementation

**Description:**
The current pull request proposes the implementation of an AppSettings Linter. This feature aims to improve the reliability and overall quality of application configuration files.

**Motivation:**
The configuration files play a crucial role in defining how applications behave in various environments. However, with increasing complexity, it's becoming challenging to keep track of all the setting definitions and potential errors. This often leads to unexpected issues, and difficulties in debugging.

**Proposed Solution:**
The AppSettings Linter is designed to tackle these challenges head-on. It will be a tool that scans the appsettings.json files for potential issues, inconsistencies, and adherence to the Stratio Vault Library syntax requirements. The linter will ensure that the configurations are correctly structured, appropriately typed, and free from common mistakes that might lead to runtime errors or security risks.

**Implementation Details:**
The AppSettings Linter is developed as a standalone utility using python. It can be integrated into CI/CD pipelines and run automatically during code reviews and deployments. A Docker image will also be implemented to improve an easier adoption of this tool.

**Testing:**
To ensure the reliability of the linter, comprehensive unit tests and integration tests will be created to cover various scenarios and edge cases.

**Expected Impact:**
By implementing the AppSettings Linter, the following benefits are expected to be achieved:

- Increased confidence in configuration correctness and consistency.
- Reduced risk of runtime errors due to misconfiguration.
- Improved security posture by identifying and addressing potential vulnerabilities.
- Streamlined code reviews, making it easier to maintain the integration of this library.

Thank you for your time and consideration. Let's work together to make this project even more robust and reliable!

### CODE REVIEW

File: utils/linter/src/validator/validator.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+                    self.appsettings_data["Vault"]["vaultAddress"],
+                    "is NOT a valid Vault address."
+                )
+
+        # Validate the syntax of the Vault mountpoint
+        if "mountPoint" in self.appsettings_data["Vault"]:
+            self.match_field(
+                self.appsettings_file,
+                self.appsettings_data["Vault"]["mountPoint"],
+                r'^(?!.*--)(?!.*\/\/)(?!.*-\/)(?!.*\/-)(?!-.*)(?!\/.*)([a-zA-Z0-9-]*\/)*[a-zA-Z0-9-]+$',
```
[the remark this answers is not in the document]
Comment (lucid-at-dream) [reply utils/linter/src/validator/validator.py:233]: I vote for spliting the handlers in functions that have an annotation and are dinamically loaded via the annotation. I think that would ease writing unit tests for each validator.

In any case, if you decide to keep this structure please add test coverage specifically to the mount point validation.

**Review by lucid-at-dream — CHANGES_REQUESTED** [review]

File: utils/linter/tests/resources/appsettings.AppRole.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "RawData": {
+    "BasePath": "\\\\192.168.1.123\\rawData\\rawData"
+  },
+  "Vault": {
+    "roleIdPath": "{% user_home %}/.vault/secrets/approle.role_id",
+    "secretIdPath": "{% user_home %}/.vault/secrets/approle.secret_id",
+    "vaultAddress": "https://vault-2022.stratio.local:8200",
```
[the remark this answers is not in the document]
Comment (rgcouto) [reply utils/linter/tests/resources/appsettings.AppRole.json:8]: Wow, this must have skipped my "anonymization" process :)
